activejob/test/cases/test_helper_test.rb

```diff
@@ -521,3 +522,9 @@ def test_assert_job_has_custom_queue_adapter_set
     assert_instance_of CustomQueueAdapter, HelloJob.queue_adapter
   end
 end
+
+class InheritedJobTest < ActiveJob::TestCase
+  def test_queue_adapter_is_test_adapter
+    assert_instance_of ActiveJob::QueueAdapters::TestAdapter, InheritedJob.queue_adapter
```

Comment:
We reset the inline adapter to test adapter here https://github.com/rails/rails/blob/51c53e19bfbc34c2b3b748786d8b5cc268de1c51/activejob/lib/active_job/test_helper.rb#L19